code/modules/mob/living/basic/space_fauna/giant_spider/spider_abilities/hivemind.dm

```diff
@@ -0,0 +1,70 @@
+/**
+ * Sets a directive to be given to all future spiders created by the user.
+ * This will be overwritten if used again.
+ */
+/datum/action/set_spider_directive
+	name = "Set Directive"
+	desc = "Set a directive for your future children to follow."
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "directive"
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	check_flags = AB_CHECK_CONSCIOUS
+	/// Current directive to apply
+	var/current_directive
+
+/datum/action/set_spider_directive/Trigger(trigger_flags)
+	. = ..()
+	if(!.)
+		return
+	current_directive = tgui_input_text(owner, "Enter the new directive", "Create directive", "[current_directive]")
```

Comment:
not sure how i feel about mutating current_directive directly like this in case of tgui_input_text schenanigans, have a local variable that we set it to after we nullcheck the results of the prompt. like
```txt
var/new_directive = tgui_input_text(...)
if(isnull(new_directive) || ...)
    return FALSE
current_directive = new_directive
```

juuuust to be super safe with handling. leaves more of a code footprint but i think it's worth it so we can "type" current_directive to a string as well rather than potentially null it.